File: src/main/java/org/trp/shincolle/block/entity/IWaypoint.java
package org.trp.shincolle.block.entity;

import net.minecraft.core.BlockPos;

import java.util.UUID;

public interface IWaypoint {
    BlockPos getLastPos();
    void setLastPos(BlockPos pos);
    BlockPos getNextPos();
    void setNextPos(BlockPos pos);
    BlockPos getChestPos();
    void setChestPos(BlockPos pos);
    UUID getOwnerUUID();
    default String getStayTimeDisplay() { return ""; }
    default String getOwnerName() { return ""; }
    default boolean showBaseParticle() { return true; }
}

File: src/main/java/org/trp/shincolle/block/entity/IWaypoint.kt
package org.trp.shincolle.block.entity

import net.minecraft.core.BlockPos
import java.util.*

interface IWaypoint {
    var lastPos: BlockPos?
    var nextPos: BlockPos?
    var chestPos: BlockPos?
    val ownerUUID: UUID?
    val stayTimeDisplay: String?
        get() = ""
    val ownerName: String?
        get() = ""

    fun showBaseParticle(): Boolean {
        return true
    }
}
